code/game/machinery/kitchen/cooking_machines/oven.dm

```diff
@@ -46,33 +46,28 @@
 		icon_state = "ovenopen"
 	..()
 
-
 /obj/machinery/appliance/cooker/oven/AltClick(var/mob/user)
-	.=1
-	if(user.stat || user.restrained())	return
-	user.setClickCooldown(DEFAULT_ATTACK_COOLDOWN)//No spamming the door, it makes a sound
-	toggle_door()
-
+	try_toggle_door(user)
+	user.setClickCooldown(DEFAULT_ATTACK_COOLDOWN)
 
 /obj/machinery/appliance/cooker/oven/verb/toggle_door()
-	set src in view()
+	set src in orange(1)
```

Comment:
`orange` is rarely the correct use, `oview` please. Range doesn't consider visibility.